### PR TITLE
geometry: clean up TangentSpaceMesh

### PR DESCRIPTION
**Pull request by poweifeng**
- Reordered the 'const var' declarations to 'var const'
 - Use std::vector for data arrays instead of manually managing allocations

### CODE REVIEW

File: libs/geometry/src/TangentSpaceMeshInternal.h [hunk 1]
```diff
@@ -22,11 +22,42 @@
...
 namespace filament::geometry {
 
 using namespace filament::math;
 using Algorithm = TangentSpaceMesh::Algorithm;
 
+template<typename T>
+class InternalArray {
+public:
+    void borrow(const T* ptr) noexcept {
+        mBorrowed = ptr;
```
Comment (pixelflinger) [inline libs/geometry/src/TangentSpaceMeshInternal.h:36]: Should it be an error to borrow a non-empty vector? Maybe we at least need an assert?

File: libs/geometry/src/TangentSpaceMeshInternal.h [hunk 2]
```diff
@@ -22,11 +22,42 @@
...
+    }
+
+    T* allocate(size_t size) noexcept {
```
Comment (pixelflinger) [inline libs/geometry/src/TangentSpaceMeshInternal.h:39]: Should it be an error to allocate() a borrowed vector? assert?

File: libs/geometry/src/TangentSpaceMeshInternal.h [hunk 3]
```diff
@@ -22,11 +22,42 @@
...
+
+    explicit operator bool() const noexcept {
+        return !mBorrowed && mAllocated.size() > 0;
+    }
+
+    const T* get() {
+        if (mBorrowed) {
+            return mBorrowed;
+        }
+        return mAllocated.data();
```
Comment (pixelflinger) [inline libs/geometry/src/TangentSpaceMeshInternal.h:53]: Here you should assert() that the vector is not empty.

File: libs/geometry/src/TangentSpaceMeshInternal.h [hunk 4]
```diff
@@ -22,11 +22,42 @@
...
+    }
+
+private:
+    std::vector<T> mAllocated;
+    const T* mBorrowed = nullptr;
```
Comment (pixelflinger) [inline libs/geometry/src/TangentSpaceMeshInternal.h:58]: `T const*` :-)

File: libs/geometry/src/TangentSpaceMeshInternal.h (hunk 1 above)
Comment (poweifeng) [reply libs/geometry/src/TangentSpaceMeshInternal.h:39]: done

File: libs/geometry/src/TangentSpaceMeshInternal.h (hunk 3 above)
Comment (poweifeng) [reply libs/geometry/src/TangentSpaceMeshInternal.h:58]: done

File: libs/geometry/src/TangentSpaceMeshInternal.h (hunk 4 above)
Comment (poweifeng) [reply libs/geometry/src/TangentSpaceMeshInternal.h]: done

File: libs/geometry/src/TangentSpaceMeshInternal.h (hunk 2 above)
Comment (poweifeng) [reply libs/geometry/src/TangentSpaceMeshInternal.h]: done

**Review by pixelflinger — APPROVED** [review]

**Pull request merged**